### DISCUSSION TITLE
Independent history per panel?

### DISCUSSION THREAD

**Discussion by nerdyman** [kovidgoyal/kitty#3895] [Q&A] [answered]
Hi, first I'd just like to say thanks for the amazing work on Kitty!

I was wondering if there's an option to enable independent histories per terminal panel? I've grown used to quitting running tasks when developing and pressing the up arrow to quickly run whatever the last command was in that panel. Kitty currently keeps track of the last command used independently of the current panel which means I often run an unintended command which I executed from another panel due to muscle memory.

Also, apologies if this has already been covered, I couldn't find any existing discussions.

**Comment by kovidgoyal** [comment, marked as answer]
history is a function of the shell, such as bash or zsh not the terminal
emulator.

On Mon, Aug 02, 2021 at 02:23:45PM -0700, Ricky wrote:
> Hi, first I'd just like to say thanks for the amazing work on Kitty!
> 
> I was wondering if there's an option to enable independent histories per terminal panel? I've grown used to quitting running tasks when developing and pressing the up arrow to quickly run whatever the last command was in that panel. Kitty currently keeps track of the last command used independently of the current panel which means I often run an unintended command from I executed from another panel due to muscle memory.
> 
> Also, apologies if this has already been covered, I couldn't find any existing discussions.
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/3895

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by nerdyman** [reply]
I see, thanks for the info!
